src/VisualStudio/Xaml/Impl/XamlVisualStudio.csproj

```diff
@@ -0,0 +1,118 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <!-- Explicitly set the language before anything else is imported. That way VSL.Settings.targets sees the right value
+         even in WPF's build of a temporary project, where it won't be able to determine it based on the file extension. -->
+    <ProjectLanguage>CSharp</ProjectLanguage>
+  </PropertyGroup>
```

Comment:
If you want to play with getting rid of the need for it, be my guest, but I spent 3 or 4 days on in in the early days of Roslyn and eventually conceded defeat to MSBuild.